tests/Microsoft.Identity.Test.Unit/TelemetryTests/RegionalTelemetryTests.cs

```diff
@@ -111,7 +113,13 @@ public async Task TelemetrySerializedTokenCacheTestAsync()
         public async Task TelemetryAutoDiscoveryFailsTestsAsync()
         {
             Trace.WriteLine("Acquire token for client with region detection fails.");
-            _harness.HttpManager.AddMockHandlerContentNotFound(HttpMethod.Get, TestConstants.ImdsUrl);
+
+            // Add mock responses for retry attempts (initial request + retries)
+            for (int i = 0; i < 3; i++)

```

Comment:
You removed the 404 from the retry policy, so you shouldn't expect MSAL to call ests 3 times.